celery/loaders/base.py

```diff
@@ -253,20 +253,26 @@ def find_related_module(package, related_name):
     # Django 1.7 allows for specifying a class name in INSTALLED_APPS.
     # (Issue #2248).
     try:
+        # Return package itself when no related_name.
         module = importlib.import_module(package)
         if not related_name and module:
             return module
-    except ImportError:
+    except ModuleNotFoundError:
+        # On import error, try to walk package up one level.
         package, _, _ = package.rpartition('.')
         if not package:
             raise
 
     module_name = f'{package}.{related_name}'
 
     try:
+        # Try to find related_name under package.
         return importlib.import_module(module_name)
-    except ImportError as e:
+    except ModuleNotFoundError as e:
         import_exc_name = getattr(e, 'name', module_name)
-        if import_exc_name is not None and import_exc_name != module_name:
```

Comment:
can we make sure this change / removal won't create any regression?

Reply:
Will do -- I should have some more time to look at this next week.

Reply:
will appreciate this to be done today.... but if you need some more time that is completely fine and we can merge this after 5.3.6 release of celery?

Reply:
Dropped an integration test in -- LMK how that looks :pray:

Reply:
that look good!! thanks